vision_layer/vision_tasks/CMakeLists.txt

```diff
@@ -37,29 +37,33 @@ include_directories(
   include/
 )
 
-add_library(buoy_task SHARED src/buoy.cpp include/buoy.h)
-add_dependencies(buoy_task ${vision_tasks_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
-target_link_libraries(buoy_task ${catkin_LIBRARIES} ${OpenCV_LIBRARIES} vision_commons )
+add_library(base_class SHARED src/base_class.cpp include/base_class.h)
```

Comment:
To simplify your CMakeLists, use glob to get all the files instead of writing them down one by one. See: https://stackoverflow.com/questions/34863374/how-to-use-cmake-file-glob-srcs-with-a-build-directory

Reply:
Yeah, I'll be changing this across the entire repository. Have started including it in the newer package.